### DISCUSSION TITLE
How to deal with the new userspace design of QMK

### DISCUSSION THREAD

**Discussion by nodupe** [manna-harbour/miryoku#267] [Q&A] [answered]
Hi, 

Since this [commit](https://github.com/qmk/qmk_firmware/pull/22222) was merged in the QMK master branch, the way of handling userspace in qmk changed. All userspaces that used to live on the master branch, including miryoku, are now deleted from it.

There is a way to access the old userspaces on the qmk-firmware repository with the use of tag "user-keymaps-still-present" but it adds some complexity to the compilition of miryoku layout, specially on a fresh clone of QMK. Not everyone who has a basic git understanding will be able to easily understand what to do.

For sure the miryoku documentation will need some adaptation.

Now, what could be the impact of such change for miryoku? Is there a strategy to handle this new way of working? 

Right now it is possible to just clone the qmk-miryoku repository and go on with our life, but I am not sure how the repo will be impacted by merging new versions of qmk code.

I've found some documentation on how to use the new way of applying userspace ( with a segregated repo ) [here](https://docs.qmk.fm/#/newbs_external_userspace).

**Comment by nodupe** [comment]
Hi,
Drash was kind enough to create a for of qmk_userspace with miryoku related configurations on it. It can be found [here](https://github.com/drashna/qmk_userspace.git).

**Comment by manna-harbour** [comment, marked as answer]
See https://github.com/manna-harbour/miryoku/discussions/287.
